game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProPurchaseUtils.java

```diff
@@ -305,9 +305,6 @@ private static ProductionRule getProductionRule(
       final Territory t, final Map<Territory, ProPurchaseTerritory> purchaseTerritories) {
 
     final List<Unit> placeUnits = new ArrayList<>();
-    if (purchaseTerritories == null) {
```

Comment:
This null check was needless?

Reply:
It was needed when ProBattleUtils.territoryHasLocalNavalSuperiority() was being called with a null param in one place, but this PR is changing that arg to an empty collection, which results in the same logic.

Reply:
Excellent  :+1: 
Thanks for clarifying.